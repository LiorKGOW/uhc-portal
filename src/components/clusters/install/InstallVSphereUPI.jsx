import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { PageSection } from '@patternfly/react-core';

import { AppPage } from '~/components/App/AppPage';

import { tollboothActions } from '../../../redux/actions';
import Breadcrumbs from '../../common/Breadcrumbs';
import PageTitle from '../../common/PageTitle';

import instructionsMapping from './instructions/instructionsMapping';
import OCPInstructions from './instructions/OCPInstructions';

export class InstallVSphereUPI extends Component {
  componentDidMount() {
    const { dispatch } = this.props;
    dispatch(tollboothActions.createAuthToken());
  }

  render() {
    const { token } = this.props;
    const breadcrumbs = (
      <Breadcrumbs
        path={[
          { label: 'Clusters' },
          { label: 'Cluster Type', path: '/create' },
          { label: 'VMware vSphere', path: '/install/vsphere' },
          { label: 'User-provisioned infrastructure' },
        ]}
      />
    );
    return (
      <AppPage title="Install OpenShift 4 | Red Hat OpenShift Cluster Manager | vSphere User-Provisioned Infrastructure">
        <PageTitle title={instructionsMapping.vsphere.upi.title} breadcrumbs={breadcrumbs} />
        <PageSection>
          <OCPInstructions
            token={token}
            cloudProviderID="vsphere"
            isUPI
            {...instructionsMapping.vsphere.upi}
          />
        </PageSection>
      </AppPage>
    );
  }
}

InstallVSphereUPI.propTypes = {
  token: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({ token: state.tollbooth.token });

export default connect(mapStateToProps)(InstallVSphereUPI);
