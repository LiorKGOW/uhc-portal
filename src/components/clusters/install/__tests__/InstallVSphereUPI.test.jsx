import React from 'react';
import { shallow } from 'enzyme';

import { InstallVSphereUPI } from '../InstallVSphereUPI';

describe('InstallVSphereUPI', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<InstallVSphereUPI token={{}} dispatch={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
