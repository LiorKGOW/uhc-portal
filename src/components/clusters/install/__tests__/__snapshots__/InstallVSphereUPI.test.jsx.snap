// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstallVSphereUPI renders correctly 1`] = `
<Fragment>
  <PageTitle
    breadcrumbs={
      <Breadcrumbs
        path={
          Array [
            Object {
              "label": "Clusters",
            },
            Object {
              "label": "Create",
              "path": "/create",
            },
            Object {
              "label": "VMware vSphere",
              "path": "/install/vsphere",
            },
            Object {
              "label": "User-provisioned infrastructure",
            },
          ]
        }
      />
    }
    title="Install OpenShift on vSphere with user-provisioned infrastructure"
  />
  <PageSection>
    <OCPInstructions
      channel="stable"
      cloudProviderID="vmware"
      customizations="https://docs.openshift.com/container-platform/latest/installing/installing_vsphere/installing-vsphere-installer-provisioned-customizations.html"
      displayRHCOSSection={true}
      docURL="https://docs.openshift.com/container-platform/latest/installing/installing_vsphere/installing-vsphere.html"
      installer="X86INSTALLER"
      isBMIPI={false}
      rhcosDownloads={
        Array [
          Object {
            "buttonText": "Download RHCOS OVA",
            "name": "OCP-Download-RHCOS-OVA",
            "url": "https://mirror.openshift.com/pub/openshift-v4/dependencies/rhcos/latest/latest/rhcos-vmware.x86_64.ova",
          },
        ]
      }
      rhcosLearnMoreURL="https://docs.openshift.com/container-platform/latest/installing/installing_vsphere/installing-vsphere.html#installation-vsphere-machines_installing-vsphere"
      title="Install OpenShift on vSphere with user-provisioned infrastructure"
      token={Object {}}
    />
  </PageSection>
</Fragment>
`;
